internal/impl/qdrant/integration_test.go

```diff
@@ -0,0 +1,227 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package qdrant
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	pb "github.com/qdrant/go-client/qdrant"
+	"github.com/redpanda-data/benthos/v4/public/service/integration"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go/modules/qdrant"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+const (
+	collectionName = `redpanda`
+	template       = `
+output:
+  label: 'qdrant'
+  qdrant:
+    grpc_host: 'localhost:$PORT'
+    tls: {enabled: false}
+    id: 'root = $POINT_ID'
+    collection_name: $COLLECTION_NAME
+    vector_mapping: 'root = $VECTOR'
+    payload_mapping: 'root = $PAYLOAD'
+`
+)
+
+func TestIntegrationQdrant(t *testing.T) {
+	integration.CheckSkip(t)
+
+	t.Parallel()
+
+	ctx := context.Background()
+	qdrantContainer, err := qdrant.Run(ctx, "qdrant/qdrant:v1.10.1")
+	require.NoError(t, err, "failed to start container")
+
+	testCases := []struct {
+		name    string
+		pointID string
+		vector  string
+	}{
+		{
+			name:    "Test With default dense vector",
+			pointID: `1`,
+			vector:  `[0.352,0.532,0.532]`,
+		},
+		{
+			name:    "Test With sparse vector",
+			pointID: `2`,
+			vector:  `{"some_sparse": {"indices":[23,325,532],"values":[0.352,0.532,0.532]}}`,
+		},
+		{
+			name:    "Test With multi vector",
+			pointID: `3`,
+			vector:  `{"some_multi": [[0.352,0.532,0.532],[0.352,0.532,0.532]]}`,
+		},
+		{
+			name:    "Test With dense and sparse vector",
+			pointID: `"465213dd-3f11-4534-8daf-9fedf203549a"`,
+			vector:  `{"some_dense": [0.352,0.532,0.532],"some_sparse": {"indices": [23,325,532],"values": [0.352,0.532,0.532]}}`,
+		},
+	}
+
+	containerPort, err := qdrantContainer.MappedPort(ctx, "6334/tcp")
+	require.NoError(t, err, "failed to get container port")
+
+	addr, err := qdrantContainer.GRPCEndpoint(ctx)
+	require.NoError(t, err, "failed to get container grpc endpoint")
+
+	err = setupCollection(ctx, addr, collectionName)
+	require.NoError(t, err, "failed to setup collection")
+
+	payload := map[string]any{
+		"str":    "str_value",
+		"number": 42,
+		"bool":   true,
+		"array":  []any{13, "str"},
+		"nested": map[string]any{
+			"nested_str": "nested_str_value",
+			"nested_num": 13,
+		},
+	}
+
+	payloadBytes, err := json.Marshal(payload)
+	require.NoError(t, err, "failed to marshal payload")
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+
+			defaultQuery := func(ctx context.Context, testID, messageID string) (string, []string, error) {
+				return fmt.Sprintf(`{"content":"%v","id":%v}`, "hello world", messageID), nil, err
+			}
+
+			suite := integration.StreamTests(
+				// Is it possible to test output only without a `getFn GetMessageFunc` arg?
```

Comment:
The idea of the `getFn` is to have a way to manually fetch whatever data the output received and then have the test suite check that whatever data was sent to the output matches what was written to it, but the assumption was that the messages are kind of stored as-is, which isn't the case here as far as I can tell.

Since your test doesn't really fit into this pattern, you can also ditch the `integration.StreamTests` utility and just spin up a stream yourself which lets you push arbitrary data into it and then again read whatever the output received and assert it. See for example [this one](https://github.com/redpanda-data/connect/blob/2bc199978b94a012fbe1422052b3524839148f71/internal/impl/gcp/integration_pubsub_test.go#L132-L231) where we just check if the output rejects the message. In your case, you can also run `assertPoints()` after the stream terminates if you follow this pattern.

Reply:
Refactor the `getFn` to assert value after write.
[`c02af75` (#2772)](https://github.com/redpanda-data/connect/pull/2772/commits/c02af75475c7fa572b4e6b40b1744b282d3eafbe)